notebooks/GPT-4V/RAG_chatcompletions_example_restapi.ipynb

```diff
@@ -0,0 +1,157 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "759f9ec0",
+   "metadata": {},
+   "source": [
+    "<h1 align =\"center\"> REST API RAG Samples</h1>\n",
+    "<hr>\n",
+    "   \n",
+    "# Chat Completions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f4b3d21a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "import os\n",
+    "import base64\n",
+    "from IPython.display import Image, display\n",
+    "from pathlib import Path\n",
+    "from shared_functions import call_GPT4V_image"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5b2d4a0f",
+   "metadata": {},
+   "source": [
+    "### Setup Parameters\n",
+    "\n",
+    "\n",
+    "Here we will load the configurations from _config.json_ file to setup vision_api_key, vision_api_endpoint, search_service_endpoint, search_index_name, and search_query_key"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fd85fb30",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load config values\n",
+    "with Path(r\"config.json\").open() as config_file:\n",
+    "    config_details = json.load(config_file)\n",
+    "\n",
+    "# Setting up the vision resource key\n",
+    "vision_api_key = os.getenv(\"VISION_API_KEY\")\n",
```

Comment:
Potential Bug: Was this intentional? VISION_API_KEY appears in config.json, but is still read from the environment.

Reply:
We keep all keys in environment variables rather than in a configuration file to offer an additional layer of security.